apps/sms/js/subject_composer.js

```diff
@@ -62,6 +62,13 @@
     }
   }
 
+  function updateDirection(value) {
+    /* jshint validthis: true */
+    var priv = privateMembers.get(this);
+
+    priv.input.dir = !value ? '': 'auto';
+  }
```

Comment:
if you agree, we can move this fix in a separate bug and land it in v2.2 only. Hopefully the gecko issue will be fixed for next version (and otherwise we'll land the same fix).